mt-kahypar/partition/preprocessing/community_detection/local_moving_modularity.h

```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+ * This file is part of KaHyPar.
+ *
+ * Copyright (C) 2019 Lars Gottesbüren <lars.gottesbueren@kit.edu>
+ * Copyright (C) 2020 Tobias Heuer <tobias.heuer@kit.edu>
+ *
+ * KaHyPar is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * KaHyPar is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with KaHyPar.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+
+#pragma once
+
+
+
+#include "mt-kahypar/datastructures/sparse_map.h"
+
+#include "mt-kahypar/macros.h"
+#include "mt-kahypar/datastructures/graph.h"
+#include "mt-kahypar/partition/context.h"
+#include "mt-kahypar/parallel/atomic_wrapper.h"
+#include "mt-kahypar/utils/randomize.h"
+
+#include "gtest/gtest_prod.h"
+
+namespace mt_kahypar::metrics {
+  double modularity(const Graph& graph, ds::Clustering& communities);
```

Comment:
should we move this into metrics.h

Reply:
moving this here was an effort to not expose internals of community detection to the rest of the framework